components/br-stream/src/router.rs

```diff
@@ -343,11 +349,14 @@ impl RouterInner {
     pub async fn do_flush(&self, task_name: &str, store_id: u64) {
         debug!("backup stream do flush"; "task" => task_name);
         if let Some(task_info) = self.tasks.lock().await.get(task_name) {
-            if let Err(e) = task_info.do_flush(store_id).await {
-                warn!("backup steam do flush fail"; "err" => ?e);
+            if task_info.set_flushing_status_cas(false, true).is_ok() {
+                if let Err(e) = task_info.do_flush(store_id).await {
+                    warn!("backup steam do flush fail"; "err" => ?e);
+                }
             }
             // set false to flushing whether success or fail
             task_info.set_flushing_status(false);
```

Comment:
If we failed to set the flushing status as `true`, we shouldn't do this. (Or there may be many flushing tasks executed concurrently.)

Reply:
That is, assuming... A flush was triggered by its size exceeds the 128MB limit...

- `task_info.flushing` would be set to `true`.
- because this change...
<img width="748" alt="image" src="https://user-images.githubusercontent.com/36239017/150783637-477074ba-929b-4738-8fbc-42fc9f9caf7d.png">

- *every* `ApplyEvent` would trigger a flush... 
- ...But it is ok, because it can be hijacked by the logic in `TaskInfo::do_flush`:
<img width="570" alt="image" src="https://user-images.githubusercontent.com/36239017/150783983-82c37adb-d8b7-43e3-b363-ad4974122094.png">

- However, with this change, I'm afraid that we would set `task_info.flushing` to `false` direct once the second flush triggered...
- ...Hence when the third time flush triggered, the guard at `TaskInfo::do_flush` won't take effect any more, making an extra flush. (Which is strange...)

Reply:
fixed